Repository/PhpcrRepository.php

```diff
@@ -119,4 +119,36 @@ protected function buildCollection(array $nodes)
 
         return $collection;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function removeNodes($nodes)
+    {
+        foreach ($nodes as $node) {
+            $node->remove();
+        }
+
+        $this->session->save();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function moveNodes($nodes, $sourceQuery, $targetPath)
+    {
+        $this->doMoveNodes($nodes, $sourceQuery, $targetPath);
+        $this->session->save();
+    }
+
+    private function doMoveNodes($nodes, $sourceQuery, $targetPath)
```

Comment:
Should be safe to type-hint on `array` here as the `finder` seems to only return arrays of things.